### ISSUE TITLE
Feedback: Introduction to Machine Learning

### ISSUE THREAD

**Issue by cabergh** [whitead/dmol-book#82]
- In 2. Introduction to Machine Learning: “Rather than rewrite this…” -> “Rather than rewriting this…”

- In 2. Introduction to Machine Learning: “...introduce the main ideas focused on a chemistry example…” -> “...applied on a chemistry example…”?

- In 2.1 The Ingredients: “...learning models **from** data...”?

- I think the descriptions in 2.1 would be clearer with complete sentences. Perhaps the students would appreciate an additional sentence explaining what it is in plain language or an example, e.g. “The features are represented as a set of N vectors of dimension D. These can be real numbers, integers, classes etc. as long as they capture the important aspects of the data, e.g. all internal distances in a molecule”.

- Didn’t try a lot to resolve this, but I’m getting dependency issues when trying to run the notebook (and using the tip): “ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
google-colab 1.0.0 requires tornado~=5.1.0; python_version >= "3.0", but you have tornado 6.1 which is incompatible.
datascience 0.10.6 requires folium==0.2.1, but you have folium 0.8.3 which is incompatible.”

- In 2.3.2 Data Exploration: spelling mistake: “...kernel density estimate overlain.” -> “...overlaid.”

- In 2.3.2 Data Exploration: “...solubility varies from about 13 to 2.5…”, should it be -13 here?

- In 2.3.2 Data Exploration: I think it would be nice to write the unit of solubility somewhere to make it clearer what the solubility numbers mean.

- In 2.3.5 Gradient Descent: “...is learning rate, which an adjustable but not trained parameter…” -> “...which is an adjustable…”.

- In 2.3.5 Gradient Descent: “Typically, it’s chosen to be some power of 10 that is at most 0.1”. Why is this? A short explanation would be nice.

- In 2.3.6 Batching: “This is making good progress.” sounds a bit strange. Perhaps “You’re/we’re making good progress” is better? Also think you shouldn’t start a sentence with “but”. Better to merge the two first sentences?

- In 2.3.7 ML Trick: Standardize features: Features with capital F. I also think you can remove “ML Trick:” from the title.

- In 2.3.8 Analyzing Model Performance: “assesssment” spelled with 3 s

- In 2.3.8 Analyzing Model Performance: “In regression, correlation coefficient is typically reported in addition to loss.” -> “...a correlation coefficient...”

- In 2.3.8 Analyzing Model Performance: “A correlation coefficient of 0.65 is OK, but not great.” Why? Perhaps mention that 1 represents the best model?

- In 2.4.1. Clustering: Perhaps nit-picking, but I think low numbers should be written with letters and not numbers in text, i.e. “2 dimensions” -> “two dimensions”

- In 2.4.1 Clustering: The examples use words like “embedding” and “manifold”, but these were never explained. The examples would be clearer if the terms were explained. Also, it’s not explained why Isomap is used instead of e.g. PCA. Some intuition of what the axes mean would be nice (e.g. for PCA it would be highest variance).

- In 2.4.1 Clustering: “...intuit what is the cluster number.” -> “...what the number of clusters should be.”

- In 2.4.1 Clustering: In the third cell, what’s meant by “whole features”?

- In 2.4.2 Choosing Cluster Number: I think “Choosing the Number of Clusters” would be better, otherwise, it sounds like some kind of (additional) special hyperparameter for clustering (the cluster number). The first sentence could also be clearer: “How do we know we had the correct number?” -> “...correct number of clusters”.

- In 2.4.2 Choosing Cluster Number: “The k-means method can give you the mean square distance from cluster center...”. Distance between the data point and the cluster center? 

- In 2.4.2 Choosing Cluster Number: “The elbow plot method is one of many approaches to selecting k[PDN05].” I don’t think the cluster number was ever introduced as k?

**Comment by awoonor** [comment]
Section 2.3.2. Data Exploration:
"The histogram shows that the solubility varies from about 13 to 2.5" should read "The histogram shows that the solubility varies from about **-13** to 2.5".

Section 2.4.1. Clustering:
Paragraph 1: "This is also the reason that it's become ..." should read "This is also the reason **why** it has become ..."
